openstudiocore/src/energyplus/Test/ElectricEquipmentITEAirCooled_GTest.cpp

```diff
@@ -0,0 +1,477 @@
+#include "../../model/CurveBiquadratic_Impl.hpp"
+#include "../../model/CurveQuadratic.hpp"
+#include "../../model/CurveQuadratic_Impl.hpp"
+#include "../../model/AirLoopHVAC.hpp"
+#include "../../model/AirLoopHVACZoneSplitter.hpp"
+#include "../../model/AirTerminalSingleDuctConstantVolumeNoReheat.hpp"
+#include "../../model/AirTerminalSingleDuctVAVReheat.hpp"
+#include "../../model/CoilHeatingElectric.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/PortList.hpp"
+#include "../../model/PortList_Impl.hpp"
+#include "../../model/ScheduleCompact.hpp"
+#include "../../model/StraightComponent.hpp"
+
+#include "../../model/LifeCycleCost.hpp"
+#include "../../model/LifeCycleCost_Impl.hpp"
+
+#include "../../utilities/idf/WorkspaceExtensibleGroup.hpp"
+#include "../../utilities/geometry/Point3d.hpp"
+
+#include <utilities/idd/ElectricEquipment_ITE_AirCooled_FieldEnums.hxx>
+#include <utilities/idd/Zone_FieldEnums.hxx>
+#include <utilities/idd/ZoneList_FieldEnums.hxx>
+#include <utilities/idd/ComponentCost_LineItem_FieldEnums.hxx>
+#include <utilities/idd/LifeCycleCost_NonrecurringCost_FieldEnums.hxx>
+#include <utilities/idd/LifeCycleCost_RecurringCosts_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+#include <resources.hxx>
+
+#include <sstream>
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+
+// OS only has definition
+TEST_F(EnergyPlusFixture, ForwardTranslator_ElectricEquipmentITEAirCooledDefinition)
+{
+  Model model;
+  ElectricEquipmentITEAirCooledDefinition definition(model);
+
+  ForwardTranslator forwardTranslator;
+  Workspace workspace = forwardTranslator.translateModel(model);
+
+  std::string errors;
+  for (const LogMessage& error : forwardTranslator.errors()){
+    errors += error.logMessage() + "\n";
+  }
+  EXPECT_EQ(0u, forwardTranslator.errors().size()) << errors;
+
+  std::string warnings;
+  for (const LogMessage& warning : forwardTranslator.warnings()){
+    warnings += warning.logMessage() + "\n";
+  }
+  EXPECT_EQ(0u, forwardTranslator.warnings().size()) << warnings;
+
+  EXPECT_EQ(0u, workspace.getObjectsByType(IddObjectType::ElectricEquipment_ITE_AirCooled).size());
+
+  model.save(toPath("./ITE_translator_definition.osm"), true);
+  workspace.save(toPath("./ITE_translator_definition.idf"), true);
+
+}
+
+// OS has IT object and definition, but no space associated => supposed to not be translated
+TEST_F(EnergyPlusFixture, ForwardTranslator_ElectricEquipmentITEAirCooled_NoSpace)
+{
+  Model model;
+  ElectricEquipmentITEAirCooledDefinition definition(model);
+  ElectricEquipmentITEAirCooled electricEquipmentITEAirCooled(definition);
+
+  ForwardTranslator forwardTranslator;
+  Workspace workspace = forwardTranslator.translateModel(model);
+
+  EXPECT_EQ(0u, workspace.getObjectsByType(IddObjectType::ElectricEquipment_ITE_AirCooled).size());
+
+  model.save(toPath("./ITE_translator_NoSpace.osm"), true);
```

Comment:
@KaiyuSun comment this out after your are done using it

Reply:
Do you mean all model.save and workspace.save?

Reply:
I commented the model.save and workspace.save in the test file.